tests/cpp/plot_dashboard_stress/main.cpp

```diff
@@ -147,9 +147,14 @@ int main(int argc, char** argv) {
 
     std::vector<size_t> offsets;
     if (temporal_batch_size.has_value()) {
+// GCC wrongfully thinks that `temporal_batch_size` might be uninit, even though we've
+// literally checked for that in the line above?!
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
         for (size_t i = 0; i < num_points_per_series; i += *temporal_batch_size) {
             offsets.push_back(i);
         }
+#pragma GCC diagnostic pop
```

Comment:
this happens so often, we have it in our compiler_utils
https://github.com/rerun-io/rerun/blob/main/rerun_cpp/src/rerun/compiler_utils.hpp#L24C9-L24C44

that would make this
```suggestion
// GCC wrongfully thinks that `temporal_batch_size` is uninitialized despite being initialized upon creation.
RR_DISABLE_MAYBE_UNINITIALIZED_PUSH
        for (size_t i = 0; i < num_points_per_series; i += *temporal_batch_size) {
            offsets.push_back(i);
        }
RR_DISABLE_MAYBE_UNINITIALIZED_POP
```

The comment is also _technically_ not entirely to the point which why my suggestion adjusts that as well: `has_value` has nothing to do with uninit. This is just a member of `std::optional` which may or may not be uninitialized which means that the outcome of it would be indetermined.
That said, this is still _obviously_ bs, since temporal_batch_size comes to life like this:
```
    const auto temporal_batch_size = args.count("temporal-batch-size")
                                         ? std::optional(args["temporal-batch-size"].as<uint64_t>())
: std::nullopt;
```

Reply:
What I don't get is why this is the only case where GCC complains in this file, even though we do the exact same thing in 10 different places... :facepalm:

Reply:
gcc moves in mysterious ways